federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPIdentityStore.java

```diff
@@ -486,8 +488,13 @@ private LDAPObject populateAttributedType(SearchResult searchResult, LDAPQuery l
                     NamingEnumeration<?> enumm = ldapAttribute.getAll();
                     while (enumm.hasMoreElements()) {
                         Object val = enumm.next();
-
-                        if (val instanceof byte[]) { // byte[]
+                        
+                        if(ldapAttributeName.toLowerCase().equals("objectsid"))
```

Comment:
As I commented in the issue, I feel that using hardcoded `objectsid` here is not OK. Maybe we should add a specific SID mapper in the msad folder. Similar to what we are doing with `MSADUserAccountControlStorageMapperFactory` but for SID syntax attributes.